### PR TITLE
Fix 2.1.x/db#69081 group layer visibility range not working

### PR DESCRIPTION
**Pull request by NathanHGit**
# Description

Added a parent layer variable to the layer component which contains the properties of the layer group. When zooming, it will check its maximum and minimum values to set the visibility of the layer.

## Useful links

Ticket: [69081 - DB - Group Layer visibility range not working](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/69081)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

I tried the zoom function on a personalized map composed of:
- 1 group layer, min zoom: 4, max zoom: 10
- layer 1, belongs to the group layer, min zoom 6, max zoom 9
- layer 2, belongs to the group layer, min zoom 2, max zoom 18

## Screenshots

peek

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/safe/src/lib/components/ui/map/layer.ts [hunk 1]
```diff
@@ -800,6 +803,35 @@ export class Layer implements LayerModel {
...
+   * @returns maximum and minimum zoom
+   */
+  public getMinMaxZoom(map: L.Map) {
+    let maxZoom = 0;
+    let minZoom = 0;
+
+    if (this.parentLayer) {
+      const parentMaxZoom = this.parentLayer.maxZoom || map.getMaxZoom();
+      const parentMinZoom = this.parentLayer.minZoom || map.getMinZoom();
+      maxZoom = this.layerDefinition?.maxZoom || map.getMaxZoom();
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/ui/map/layer.ts]: Hi @NathanHGit , this two lines are repeated

```
maxZoom = this.layerDefinition?.maxZoom || map.getMaxZoom();
minZoom = this.layerDefinition?.minZoom || map.getMinZoom();
```

Do you see how can you keep just one definition of those?

pseudo code hint:

```
get zoom values from layerDefinition

if there is parentLayer
   get zoom values from parentLayer
   if zoom values are greater/lesser than the values from layerDefinition
     then take those values from parentLayer
     
return whatever zoom values are set
```

File: libs/safe/src/lib/components/ui/map/layer.ts [hunk 2]
```diff
@@ -176,6 +176,9 @@ export class Layer implements LayerModel {
   // Declare variables to store the event listeners
   private zoomListener!: L.LeafletEventHandlerFn;
 
+  // Group layer
+  public parentLayer!: any;
```
Comment (unai-reliefapp) [inline libs/safe/src/lib/components/ui/map/layer.ts:180]: I cannot see this property assigned anywhere in the code, please double check this

**Review by unai-reliefapp — CHANGES_REQUESTED** [review]
Maybe it's not finished as it's still a Draft PR, but if so please double check both comments
KR, Unai

File: libs/safe/src/lib/components/ui/map/layer.ts (hunk 2 above)
Comment (NathanHGit) [reply libs/safe/src/lib/components/ui/map/layer.ts:180]: This property is assigned at line 511 `child.parentLayer = this.layerDefinition;`

**Review by unai-reliefapp — APPROVED** [review]

**Comment by AntoineRelief** [comment]
closing this PR but I'll put the code in 2.x.x

it is mostly working

**Pull request closed without merging**